### ISSUE TITLE
aws_acm_certificate does not create new certificate for the existing domain

### ISSUE THREAD

**Issue by jimmytoan** [terraform-providers/terraform-provider-aws#7909]
### Terraform Version
* Terraform v0.11.8
* provider.aws v2.1.0

### Affected Resource(s)
* aws_acm_certificate

### Terraform Configuration Files

* This file is on Test-1 folder
```hcl
provider "aws" {
  region = "ap-southeast-2"
}

resource "aws_acm_certificate" "test-1" {
  domain_name       = "*.test.com"
  validation_method = "DNS"

  tags = {
    Name        = "test-1"
  }
}
```

* This file is on Test-2 folder (separates with TF state file for Test-1)
```hcl
provider "aws" {
  region = "ap-southeast-2"
}

resource "aws_acm_certificate" "test-2" {
  domain_name       = "*.test.com"
  validation_method = "DNS"

  tags = {
    Name        = "test-2"
  }
}
```

### Expected Behavior
Terraform should create 2 certificates (different certificate arn/id) for domain *.test.com on AWS ACM console

### Actual Behavior and Steps to Reproduce
* From Test-1 folder, run terraform apply =>  terraform creates only 1 certificate for domain *.test.com with name test-1 
* From Test-2 folder, run terraform apply => terraform update the certificate ARN above with name test-2
* This ends up with 2 TF state files include the same certificate arn

**Comment by aeschright** [comment]
Hi @jimmytoan -- thank for taking the time to report this! Can you tell us a little more about what you were trying to do and your configuration? I don't think there's enough here for us to know how to follow up.

**Comment by bflad** [comment]
Hi folks 👋 This issue was resolved awhile ago in version 2.23.0 of the Terraform AWS Provider and should have been fixed in all versions since. Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

If you are still having trouble on recent versions of the Terraform AWS Provider, please create a [new GitHub issue following the bug report template](https://github.com/terraform-providers/terraform-provider-aws/issues/new?template=Bug_Report.md). Thanks!
